components/optimizer/greedymax.py

```diff
@@ -0,0 +1,27 @@
+from .base import BaseOptimizer
+
+
+class GreedyMax(BaseOptimizer):
+    """
+    GreedyMax is an optimizer that schedules the visits in a greedy fashion.
+    """
+
+    def __init__(self, some_parameter=1):
+        # Parameters specifically for the GreedyMax optimizer should be go here
+        self.some_parameter = some_parameter
+        self._visits = []
+        self.selection = None
+    
+    def _run(self):
+        ...
+
+    def add(self, selection):
```

Comment:
What type will `selection` be? I'm curious how you're envisioning this.
I'm guessing we'll need groups and their scores as the selection information?